### PR TITLE
COMMON: More unit tests for common

### PR DESCRIPTION
**Pull request by vyzigold**
COMMON: Add several missing unit tests

Add tests for Common::rad2deg, Common::deg2rad from common/math.h
Add tests for Common::isPunct from Common/util
Add more tests for Common::String (firstChar, setChar, insertChar) from common/str

### CODE REVIEW

**Comment by Mataniko** [comment]
Thank you for your contribution Vyzigold. Can you please edit the commit messages and PR name to better match our naming conventions?

You can find the guidelines here: https://wiki.scummvm.org/index.php?title=Commit_Guidelines

**Comment by vyzigold** [comment]
@Mataniko I'm sorry for the PR name, I wanted to change it after I write the PR message, but I totaly forgot about it. I hope it's all right now.

File: test/common/math.h [hunk 1]
```diff
@@ -1,7 +1,14 @@
 #include <cxxtest/TestSuite.h>
 
 #include "common/math.h"
+/* Macro function for asserting that the compared values x and y
+ * aren't further appart from each other than z.
```
Comment (bluegr) [inline test/common/math.h]: Appart -> apart

Also, this multi-line comment formatting is not used in other places

Comment (vyzigold) [reply test/common/math.h]: Thanks for the comment, I changed the formatting to only use '//'. I hope it is all right now.

**Comment by bluegr** [comment]
Nice work! Merging

**Pull request merged**